Sources/BraveWallet/Settings/WalletSettingsView.swift

```diff
@@ -7,143 +7,62 @@ import SwiftUI
-  @ObservedObject var keyringStore: KeyringStore
-  @ObservedObject var displayDappsNotifications = Preferences.Wallet.displayWeb3Notifications
-
+public struct Web3SettingsView: View {
+  var settingsStore: SettingsStore?
+  var networkStore: NetworkStore?
+  var keyringStore: KeyringStore?
+  
+  @ObservedObject var enableSNSDomainName = Preferences.Wallet.resolveSNSDomainNames
+  
   @State private var isShowingResetWalletAlert = false
   @State private var isShowingResetTransactionAlert = false
   /// If we are showing the modal so the user can enter their password to enable unlock via biometrics.
   @State private var isShowingBiometricsPasswordEntry = false
-
+  
+  private var domainOptions: [Preferences.Wallet.Web3DomainOption] = [.ask, .enable, .disable]
+  
   public init(
-    settingsStore: SettingsStore,
-    networkStore: NetworkStore,
-    keyringStore: KeyringStore
+    settingsStore: SettingsStore? = nil,
+    networkStore: NetworkStore? = nil,
+    keyringStore: KeyringStore? = nil
   ) {
     self.settingsStore = settingsStore
     self.networkStore = networkStore
     self.keyringStore = keyringStore
   }
-
-  private var autoLockIntervals: [AutoLockInterval] {
-    var all = AutoLockInterval.allOptions
-    if !all.contains(settingsStore.autoLockInterval) {
-      // Ensure that the users selected interval always appears as an option even if
-      // we remove it from `allOptions`
-      all.append(settingsStore.autoLockInterval)
-    }
-    return all.sorted(by: { $0.value < $1.value })
-  }
-
+  
   public var body: some View {
     List {
-      Section(
-        footer: Text(Strings.Wallet.autoLockFooter)
-          .foregroundColor(Color(.secondaryBraveLabel))
-      ) {
-        Picker(selection: $settingsStore.autoLockInterval) {
-          ForEach(autoLockIntervals) { interval in
-            Text(interval.label)
-              .foregroundColor(Color(.secondaryBraveLabel))
-              .tag(interval)
-          }
-        } label: {
-          Text(Strings.Wallet.autoLockTitle)
-            .foregroundColor(Color(.braveLabel))
-            .padding(.vertical, 4)
-        }
-        .listRowBackground(Color(.secondaryBraveGroupedBackground))
+      if let settingsStore, let networkStore, let keyringStore, settingsStore.isDefaultKeyringCreated {
+        WalletSettingsView(
+          settingsStore: settingsStore,
+          networkStore: networkStore,
+          keyringStore: keyringStore,
+          isShowingResetWalletAlert: $isShowingResetWalletAlert,
+          isShowingResetTransactionAlert: $isShowingResetTransactionAlert,
+          isShowingBiometricsPasswordEntry: $isShowingBiometricsPasswordEntry
+        )
       }
-      Section {
-        Picker(selection: $settingsStore.currencyCode) {
-          ForEach(CurrencyCode.allCurrencyCodes) { currencyCode in
-            Text(currencyCode.code)
+      Section(header: Text(Strings.Wallet.web3DomainOptionsHeader)) {
```

Comment:
Only concern with this feature flag change now is that the Web3 settings will be empty if not wallet is setup (or after wallet reset). Makes it a bit annoying to implement, maybe it's easier to add IPFS preference in now to avoid adding temp logic for when to show Web3 in main app Settings?

Reply:
we will move IPFS settings (currently under Settings) to Web3 so that there won't be an empty case.